### PR TITLE
[WORKFLOWS-62 | WORKFLOWS-76] Add `CostCenter` tags

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
This PR comes after a discussion with IT on how we can best implement tagging in a way that's more useful for Finance. For more context, I suggest you read through [WORKFLOWS-76](https://sagebionetworks.jira.com/browse/WORKFLOWS-76) and [IT-796](https://sagebionetworks.jira.com/browse/IT-796). I believe these AWS accounts will be the first to implement this approach, so I'm all ears for feedback. 

You can skip to [this point](https://github.com/Sage-Bionetworks-Workflows/nextflow-infra/pull/77/files#diff-541004e9a888f44ff5dd5471545457a40141db89a4dc344a364bd83455324542) in the "Files changes" tab to see examples of program codes being used (everything else is just using a template variable). 

I'm also tackling [WORKFLOWS-62](https://sagebionetworks.jira.com/browse/WORKFLOWS-62) to reduce repetition of our default stack tags.

### CODE REVIEW

**Comment by BrunoGrandePhD** [comment]
@brucehoff: I can't request a review from you yet since you're not a member of this GitHub organization (invite sent), but I'm tagging you in case you're interested.

**Review by thomasyu888 — APPROVED** [review]
Tremendous effort @BrunoGrandePhD .  LGTM

**Review by tthyer — APPROVED** [review]

File: config/projects/amp-ad-project.yaml [hunk 1]
```diff
@@ -22,3 +22,4 @@ stack_tags:
   Department: NDR
   Project: amp-ad
   OwnerEmail: william.poehlman@sagebase.org
+  SageFinanceProgramCode: NIA AMP-AD CC | 10015
```
Comment (zaro0508) [inline config/projects/amp-ad-project.yaml:25]: would be nice to make stack_tags resolvable (i.e. https://github.com/Sceptre/sceptre/pull/847), that way we could pull all these tags in from some type of spreadsheet.

also, not a big deal, the term AWS likes to use for this is `CostCenter` https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html

**Review by zaro0508 — APPROVED** [review]

File: config/projects/amp-ad-project.yaml (hunk 1 above)
Comment (BrunoGrandePhD) [reply config/projects/amp-ad-project.yaml:25]: @brucehoff: What do you think about `SageFinanceProgramCode` vs `CostCenter`?

**Review by brucehoff — COMMENTED** [review]
The outstanding issue is that if the list of CostCenters defined by Sage changes then the tags added here will be wrong.  I would like to see a central list of CostCenters (that Sage can maintain) and a mechanism (like a Service Control Policy) to make sure that tags added to resources here conform to that list.

**Pull request merged**